lib/modules/manager/conda/extract.ts

```diff
@@ -0,0 +1,170 @@
+    // no channels that can be used. We therefore create
+    // a faulty package and return it
+    // This is only done to prevent unnecessary HTTP requests
+    if (channels.length === 0) {
+      return {
+        deps: [
+          {
+            depName: 'nodefaults',
+            skipReason: 'invalid-config',
+          },
+        ],
+      };
+    }
+
+    // We include default channels for Unix, but not msys2 which is default on Windows.
+    // If users use Windows and want all updates with defaults, they need to explicitly
+    // include msys2 as channel
+  } else {
+    // Remove 'defaults' if any channels are specified
+    if (parsedContent.channels) {
+      channels = parsedContent.channels.filter((obj) => {
+        return obj !== 'defaults';
+      });
+    }
+
+    // Always add default channels
+    channels.push('main', 'r');
+  }
+
+  const deps: PackageDependency[] = [];
+  parsedContent.dependencies?.forEach((parsedDep) => {
+    // If the parsedDep is the 'pip' key, continue
+    if (typeof parsedDep === 'object') {
+      logger.debug(
+        'Found object in package file, assuming pip key, which is unsupported'
+      );
+      return;
+    }
+
+    // Parse the dependency string
+    const match = matchSpecRegex.exec(parsedDep);
+    if (match?.groups) {
+      const dep = extractDependency(match.groups, channels);
+      if (dep) {
+        deps.push(...dep);
+      }
+    } else {
+      // If we can't parse the dependency correctly, we set the whole line as
+      // depName so that users can more easily debug what part of the dependency
+      // is not supported yet in the manager
+      deps.push({
+        skipReason: 'unsupported',
+        depName: parsedDep,
+      });
+    }
+  });
+
+  return deps.length ? { datasource: 'conda', deps } : null;
+}
+
+export function extractDependency(
+  groups: Record<string, string>,
+  channels: Array<string>
+): PackageDependency[] | null {
+  const dep: PackageDependency = {
+    // We set depName to the captured name so that on invalid or
+    // unsupported configurations it is shown as invalid/unsupported once
+    depName: groups.name,
+    currentValue: groups.version,
+  };
+
+  // To ensure that we only push correct updates, any specification containing
+  // one of the following is unsupported as of now:
+  //   * key-value pairs
+  //   * an epoch
+  //   * an exact build version (those are specified as name==version=build_version)
+  if (groups.keyvalue || groups.epoch || groups.build) {
+    dep.skipReason = 'unsupported';
+    return [dep];
+  }
+
+  // Check if the version is strict semver
+  const versionMatch = releaseVersionRegex.exec(groups.version);
+
+  // If exact matching is used (name==version), we know the exact
+  // version used and set it explicitly.
+  if (versionMatch?.groups?.version && groups.exact === '==') {
+    dep.currentVersion = groups.version;
+    dep.versioning = 'semver';
```

Comment:
i think you should use pep440 versioning instead, as it will support most of your limitations